client/my-sites/media-library/media-date-range.jsx

```diff
@@ -0,0 +1,344 @@
+					inputFromDate: this.nativeDateToMoment( newRange.from ).format( 'L' ),
+					inputToDate: this.nativeDateToMoment( newRange.to ).format( 'L' ),
+				};
+
+				// For first date selection only: "cache" previous dates
+				// just in case user doesn't "Apply" and we need to revert
+				// to the original dates
+				if ( ! this.state.oldDatesSaved ) {
+					newState = Object.assign( {}, newState, {
+						oldStartDate: previousState.startDate,
+						oldEndDate: previousState.endDate,
+						oldDatesSaved: true, // marks that we have saved old dates
+					} );
+				}
+
+				return newState;
+			},
+			() => this.props.onDateSelect( this.state.startDate, this.state.endDate )
+		);
+	}
+
+	revertDates() {
+		this.setState( previousState => {
+			const newState = { oldDatesSaved: false };
+
+			if ( previousState.oldStartDate && previousState.oldEndDate ) {
+				newState.startDate = previousState.oldStartDate;
+				newState.endDate = previousState.oldEndDate;
+			}
+
+			return newState;
+		} );
+	}
+
+	commitDates() {
+		this.setState(
+			previousState => ( {
+				oldStartDate: previousState.startDate, // update cached old dates
+				oldEndDate: previousState.endDate, // update cached old dates
+				oldDatesSaved: false,
+			} ),
+			() => {
+				this.props.onDateCommit( this.state.startDate, this.state.endDate );
+				this.togglePopover();
+			}
+		);
+	}
+
+	momentDateToNative( momentDate ) {
+		return momentDate.toDate();
+	}
+
+	nativeDateToMoment( nativeDate ) {
+		return this.props.moment( nativeDate );
+	}
+
+	dateToHumanReadable( date ) {
+		return this.props.moment( date ).format( 'L' );
+	}
+
+	renderDatePicker() {
+		const now = new Date();
+
+		return (
+			<DatePicker
+				className="media-library__date-range-popover-date-picker"
+				showOutsideDays={ false }
+				toMonth={ now }
+				onSelectDay={ this.onSelectDate }
+				selectedDays={ {
+					from: this.momentDateToNative( this.state.startDate ),
+					to: this.momentDateToNative( this.state.endDate ),
+				} }
+				numberOfMonths={ window.matchMedia( '(min-width: 480px)' ).matches ? 2 : 1 }
+				calendarViewDate={ this.momentDateToNative( this.state.startDate ) }
+				disabledDays={ [
+					{
+						after: now, // you can't look at photos from the future!
+					},
+				] }
+			/>
+		);
+	}
+
+	renderPopoverHeader() {
+		return (
+			<div className="media-library__date-range-popover-header">
+				<Button
+					className="media-library__date-range-popover-cancel-btn"
+					onClick={ this.togglePopover }
+					compact
+				>
+					Cancel
+				</Button>
+				<Button
+					className="media-library__date-range-popover-apply-btn"
+					primary
+					onClick={ this.commitDates }
+					compact
+				>
+					<Gridicon icon="checkmark" />
+					Apply Dates
+				</Button>
+			</div>
+		);
+	}
+
+	renderTriggerButton() {
+		return (
+			<Button
+				className="media-library__date-range-btn"
+				ref={ this.startButtonRef }
+				onClick={ this.togglePopover }
```

Comment:
I think this is the button intended to commit the dates right?
If so I would imagine a method somewhere along the lines of `onConfirmDateRange` (but with better naming maybe).
This would handle both closing the popover and commiting those settings.

```js
onConfirmDateRange = () => {
  doTheCommitingLogic...
  this.closePopver();
}
```
Similar to `togglePopover`s current logic, but we're separating those responsibilities, avoiding conflation of responsibilities.

Reply:
> I think this is the button intended to commit the dates right?

This button shouldn't commit the dates. It should close the popover whilst also ensuring that the _selected_ dates are reverted. 

The only time dates should be committed are when you click `Apply`. 

I think we need to name the methods more clearly or write better code comments. 

Also have you seen the tests which I hoped covered some of these scenarios?

Reply:
To confirm

1. This button is only intended to toggle the popover. It does not commit dates in any way. The only way to commit dates is via `Apply` btn
2. Way already have dedicated `commitDates` method. This seperate from `openPopover` and `closePopover` which handle open/close only.